drv/gimlet-seq-server/src/main.rs

```diff
@@ -386,6 +396,63 @@ fn main() -> ! {
     }
 }
 
+fn read_vpd() {
```

Comment:
In the PR, you mentioned disliking that this is copy-pasted into every sequencer, which I agree with!

How about moving it into the `task-packrat-api` crate, so everyone does
```rust
let i2c = I2C.get_task_id();
let packrat = Packrat::from(PACKRAT.get_task_id());
task_packrat_api::read_vpd(i2c, packrat);
```
instead?

Reply:
The problem with moving it is the `ringbuf_entry!` calls - those don't make sense from `task-packrat-api`. I could:

* remove them
* add callbacks to report errors
* remove the retry loop and just return an error, although this would complicate the caller enough that we might still have 3 moderately-sized functions copy-pasted

Reply:
Ah, I see.

You _could_ put a ringbuf into `task-packrat-api` if you wanted; having ringbufs in library crates isn't totally standard, but we do it in a few drivers already.

Reply:
(although then every task that calls _into_ packrat would get the ringbuf allocated; maybe an extra layer of indirection with a `drv-packrat-writer` crate that includes just this function and ringbuf?)